src/System.Private.CoreLib/shared/System/RuntimeType.cs

```diff
@@ -392,6 +392,32 @@ public override bool IsAssignableFrom(Type? c)
             }
 
             return RuntimeTypeHandle.GetBaseType(this);
-        }        
+        }
+
+        // This is used by the ToString() overrides of all reflection types. The legacy behavior has the following problems:
+        //  1. Use only Name for nested types, which can be confused with global types and generic parameters of the same name.
+        //  2. Use only Name for generic parameters, which can be confused with nested types and global types of the same name.
+        //  3. Remove the namespace ("System") for all primitive types, which is not language neutral.
+        //  4. MethodBase.ToString() use "ByRef" for byref parameters which is different than Type.ToString().
+        //  5. ConstructorInfo.ToString() outputs "Void" as the return type. Why Void?
+        internal override string FormatTypeName()
```

Comment:
Can this be on `System.Type` and be non-virtual?

Reply:
Done.